src/Microsoft.CodeAnalysis.Analyzers/Core/CodeAnalysisDiagnosticsResources.resx

```diff
@@ -398,4 +398,13 @@
   <data name="DoNotUseReservedDiagnosticIdTitle" xml:space="preserve">
     <value>Do not use reserved diagnostic IDs.</value>
   </data>
+  <data name="DoNotUseCompilationGetSemanticModelDescription" xml:space="preserve">
+    <value>This call leads the compiler to rebind the entire file, and doing it multiple times (say with SyntaxNode callbacks) can be a performance hog.</value>
+  </data>
+  <data name="DoNotUseCompilationGetSemanticModelMessage" xml:space="preserve">
+    <value>This call can lead to performance hog and should be used carefully.</value>
+  </data>
+  <data name="DoNotUseCompilationGetSemanticModelTitle" xml:space="preserve">
+    <value>Do not invoke Compilation.GetSemanticModel() method.</value>
+  </data>
```

Comment:
I am not happy with those messages.

Also, I am not sure whether the rule is intended to be a strict "do not use" or more of a "review use of".

Reply:
No worries, we will get the wording figured out. Thanks for the contribution!

Reply:
How about
- Title/Message: `Do not invoke Compilation.GetSemanticModel() method within a diagnostic analyzer`
- Description: `'GetSemanticModel' is an expensive method to invoke within a diagnostic analyzer because it creates a completely new semantic model, which does not share compilation data with the compiler or other analyzers. This incurs an additional performance cost during semantic analysis. Instead, consider registering a different analyzer action which allows used of a shared 'SemanticModel', such as 'RegisterOperationAction', 'RegisterSyntaxNodeAction', or 'RegisterSemanticModelAction'`.

Reply:
@Evangelink I think we should be good to merge once you update the diagnostic resource strings.